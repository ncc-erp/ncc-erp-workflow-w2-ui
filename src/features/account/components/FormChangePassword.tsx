import {
  FormControl,
  FormLabel,
  Divider,
  Heading,
  Button,
  Input,
  Flex,
  Box,
} from "@chakra-ui/react";
import { yupResolver } from "@hookform/resolvers/yup";
import { useForm } from "react-hook-form";
import { changePasswordSchema } from "utils/yup";

interface IFormChangePassword {
  currentPassword: string;
  newPassword: string;
  newPasswordConfirm: string;
}

const FormChangePassword = () => {
  const methods = useForm<IFormChangePassword>({
    resolver: yupResolver(changePasswordSchema),
    // defaultValues: {
    //   samples: [{ volume: "" }],
    // },
  });
  const {
    handleSubmit,
    register,
    formState: { errors, isSubmitting },
  } = methods;

  function onSubmit(values: IFormChangePassword) {
    console.log(values);
  }
  return (
    <>
      <Heading>Change password</Heading>
      <Divider mb={5} />
      <form onSubmit={handleSubmit(onSubmit)}>
        <FormControl>
          <Flex flexDirection={"column"} gap={5}>
            <Box>
              <FormLabel htmlFor="currentPassword">
                Current password *
              </FormLabel>
              <Input
                isInvalid={errors?.currentPassword as boolean | undefined}
                id="currentPassword"
                placeholder="Current password"
                {...register("currentPassword")}
              />
              <Box fontSize={14} color={"red.500"} position={"absolute"}>
                {errors?.newPassword &&
                  (errors?.currentPassword?.message as string)}
              </Box>
            </Box>
            <Box>
              <FormLabel htmlFor="newPassword">New password *</FormLabel>
              <Input
                isInvalid={errors?.newPassword as boolean | undefined}
                id="newPassword"
                placeholder="New password"
                {...register("newPassword")}
              />
              <Box fontSize={14} color={"red.500"} position={"absolute"}>
                {errors?.newPassword &&
                  (errors?.newPassword?.message as string)}
              </Box>
            </Box>
            <Box>
              <FormLabel htmlFor="newPasswordConfirm">
                Confirm new password *
              </FormLabel>
              <Input
                isInvalid={errors?.newPasswordConfirm as boolean | undefined}
                id="newPasswordConfirm"
                placeholder="Confirm new password"
                {...register("newPasswordConfirm")}
              />
              <Box fontSize={14} color={"red.500"} position={"absolute"}>
                {errors?.newPasswordConfirm &&
                  (errors?.newPasswordConfirm?.message as string)}
              </Box>
            </Box>
          </Flex>
          <Button
            mt={7}
            colorScheme="blue"
            isLoading={isSubmitting}
            type="submit"
          >
            Submit
          </Button>
        </FormControl>
      </form>
    </>
  );
};

export default FormChangePassword;
