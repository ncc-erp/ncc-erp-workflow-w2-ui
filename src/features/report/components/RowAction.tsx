import {
  Menu,
  MenuButton,
  IconButton,
  MenuList,
  MenuItem,
  Icon,
  useColorModeValue,
} from '@chakra-ui/react';
import { RiSettings4Fill } from 'react-icons/ri';
import { FaEye } from 'react-icons/fa';
import { ColorThemeMode } from 'common/constants';

interface IRowActionProps {
  onViewDetails: () => void;
}

export const RowAction = ({ onViewDetails }: IRowActionProps) => {
  const bg = useColorModeValue(ColorThemeMode.LIGHT, ColorThemeMode.DARK);
  const color = useColorModeValue(ColorThemeMode.DARK, ColorThemeMode.LIGHT);

  return (
    <Menu>
      <MenuButton
        as={IconButton}
        aria-label=""
        variant="ghost"
        size="sm"
        icon={<Icon color="gray.500" fontSize="lg" as={RiSettings4Fill} />}
      />
      <MenuList minW="100px" bg={bg}>
        <MenuItem
          color={color}
          display="flex"
          gap="12px"
          onClick={onViewDetails}
        >
          <Icon color="gray.500" as={FaEye} />
          View
        </MenuItem>
      </MenuList>
    </Menu>
  );
};
