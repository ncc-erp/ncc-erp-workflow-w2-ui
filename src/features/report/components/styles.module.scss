.container {
  display: flex;
  justify-content: space-between;
  margin-top: 10px;

  .btnShowMore {
    font-size: 14px;
  }

  .left,
  .right {
    width: 48%;
  }

  .dates {
    display: flex;
    flex-wrap: wrap;

    .date {
      margin-right: 5px;
      margin-bottom: 10px;
      width: 100px;
      border-radius: 5px;
      text-align: center;
      font-size: 15px;
      font-weight: 500;
      padding: 5px;
    }

    .countedDate {
      border-bottom: 4px solid var(--chakra-colors-green-700);
    }
  }
}

.btnExport {
  text-align: right;
  // margin-right: 30px;
  // margin-top: 10px;
  svg {
    color: var(--chakra-colors-gray-500);
  }
}

.badge {
  padding: 2px 8px;
  border-radius: 24px;
  font-size: 12px;
  color: #ffffff;
  font-weight: 600;
  text-align: center;
}
.Pending {
  background-color: var(--chakra-colors-blue-700);
}
.Approved {
  background-color: var(--chakra-colors-green-700);
}
.Rejected {
  background-color: var(--chakra-colors-red-700);
}
.Canceled {
  background-color: var(--chakra-colors-gray-700);
}
.Faulted {
  background-color: var(--chakra-colors-yellow-700);
}

.datepickerWrapper {
  flex: 1;
}

.datePicker {
  width: 100%;
}
