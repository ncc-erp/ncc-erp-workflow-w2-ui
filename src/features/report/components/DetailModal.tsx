import {
  Divider,
  HStack,
  Heading,
  Image,
  Modal,
  ModalBody,
  ModalCloseButton,
  ModalContent,
  ModalHeader,
  ModalOverlay,
  Text,
  useColorModeValue,
} from '@chakra-ui/react';
import { IPostAndWFH } from 'models/report';
import Logo from 'assets/images/ncc_logo.svg';
import styles from './styles.module.scss';
import { CardDetails } from 'common/components/CardDetails';
import { useState } from 'react';
import moment from 'moment';
import { ColorThemeMode } from 'common/constants';

interface IDetailModal {
  isOpen: boolean;
  onClose: () => void;
  reportDetail: IPostAndWFH;
  startDate: Date | null;
  endDate: Date | null;
}

const itemsPerPage = 2;

export const DetailModal = ({
  isOpen,
  onClose,
  reportDetail,
  startDate,
  endDate,
}: IDetailModal) => {
  const bg = useColorModeValue(ColorThemeMode.LIGHT, ColorThemeMode.DARK);
  const color = useColorModeValue(ColorThemeMode.DARK, ColorThemeMode.LIGHT);

  const [visibleCount, setVisibleCount] = useState(itemsPerPage);
  const posts = reportDetail?.posts;
  const requestDates = reportDetail?.requestDates;

  const showMore = () => {
    setVisibleCount(visibleCount + itemsPerPage);
  };

  const shouldEnableScroll = posts && posts.length >= itemsPerPage * 2;

  return (
    <Modal isOpen={isOpen} onClose={onClose} closeOnOverlayClick={false}>
      <ModalOverlay />
      <ModalContent p="10px" maxW="800px" pb={5}>
        <ModalHeader fontSize="18px">
          <HStack>
            <Image h="45px" src={Logo} />
            <Heading ml={1} w="550px">
              <Text color="primary" fontSize={18} fontWeight={600}>
                Report Details
              </Text>
              <Text fontSize={16} fontWeight={400} mt={1.5}>
                {reportDetail?.email}
              </Text>
            </Heading>
          </HStack>
        </ModalHeader>
        <ModalBody>
          <Divider mb={5}></Divider>
          <div className={styles.container}>
            <div className={styles.left}>
              <Text
                mb="15px"
                fontWeight={600}
                fontStyle="italic"
                color="primary"
              >
                Post Details
              </Text>
              <div
                style={{
                  maxHeight: shouldEnableScroll ? '450px' : 'auto',
                  overflowY: shouldEnableScroll ? 'auto' : 'visible',
                }}
              >
                {posts && (
                  <div>
                    {posts
                      .sort(
                        (a, b) =>
                          new Date(b.date).valueOf() -
                          new Date(a.date).valueOf()
                      )
                      .slice(0, visibleCount)
                      .map((item, index) => (
                        <CardDetails
                          key={index}
                          title={item.title.rendered}
                          date={item.date}
                          link={item.link}
                        />
                      ))}
                    {visibleCount < posts.length ? (
                      <button className={styles.btnShowMore} onClick={showMore}>
                        Show more...
                      </button>
                    ) : posts.length === 0 ? (
                      <Text>No posts found!</Text>
                    ) : null}
                  </div>
                )}
              </div>
            </div>
            <div className={styles.right}>
              <Text
                mb="15px"
                fontWeight={600}
                fontStyle="italic"
                color="primary"
              >
                Request Details
              </Text>
              {requestDates && requestDates.length > 0 ? (
                <div className={styles.dates}>
                  {requestDates
                    ?.filter((date) => {
                      const currentDate = moment(date, 'DD/MM/YYYY').toDate();
                      if (startDate && endDate) {
                        return (
                          currentDate >= startDate && currentDate <= endDate
                        );
                      }
                      return true;
                    })
                    .map((filteredDate) => (
                      <Text
                        key={filteredDate}
                        className={styles.date}
                        bg={bg}
                        color={color}
                      >
                        {filteredDate}
                      </Text>
                    ))}
                </div>
              ) : (
                <Text>No dates found!</Text>
              )}
            </div>
          </div>
        </ModalBody>
        <ModalCloseButton />
      </ModalContent>
    </Modal>
  );
};
