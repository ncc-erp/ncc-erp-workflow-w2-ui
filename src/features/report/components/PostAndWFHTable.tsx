import {
  Box,
  Button,
  Center,
  HStack,
  IconButton,
  Input,
  InputGroup,
  InputRightElement,
  Spacer,
  Spinner,
  Text,
  Wrap,
  WrapItem,
} from '@chakra-ui/react';
import { useWfhList } from 'api/apiHooks/reportHooks';
import { Pagination } from 'common/components/Pagination';
import { PageSize } from 'common/components/Table/PageSize';
import { ShowingItemText } from 'common/components/Table/ShowingItemText';
import { Table } from 'common/components/Table/Table';
import { noOfRows } from 'common/constants';
import { SortDirection, WfhSortField } from 'common/enums';
import { FilterWfhParams, IPostAndWFH } from 'models/report';
import { useEffect, useMemo, useRef, useState } from 'react';
import { useRecoilValue } from 'recoil';
import { appConfigState } from 'stores/appConfig';

import {
  ColumnDef,
  SortingState,
  createColumnHelper,
} from '@tanstack/react-table';
import DateRangePicker from 'common/components/DateRangePicker';
import { EmptyWrapper } from 'common/components/EmptyWrapper';
import useDebounced from 'hooks/useDebounced';
import ReactDatePicker from 'react-datepicker';
import { FaDownload } from 'react-icons/fa';
import { TbSearch } from 'react-icons/tb';
import { formatDate } from 'utils';
import { handleExportExcelFile } from 'utils/handleExportExcelFile';
import { DetailModal } from './DetailModal';
import { RowAction } from './RowAction';
import styles from './styles.module.scss';
import { AiOutlineReload } from 'react-icons/ai';

const initialFilter: FilterWfhParams = {
  maxResultCount: +noOfRows[2].value,
  skipCount: 0,
  sorting: [WfhSortField.totalMissingPosts, 'desc'].join(' '),
  keySearch: '',
  startDate: null,
  endDate: null,
};

const initialSorting: SortingState = [
  {
    id: WfhSortField.totalMissingPosts,
    desc: true,
  },
];

export const TablePostAndWFH = () => {
  const [filter, setFilter] = useState<FilterWfhParams>(initialFilter);
  const [sorting, setSorting] = useState<SortingState>(initialSorting);
  const columnHelper = createColumnHelper<IPostAndWFH>();
  const { sideBarWidth } = useRecoilValue(appConfigState);
  const { data, isLoading, isRefetching, refetch } = useWfhList(filter);
  const { items: wfhList = [], totalCount = 0 } = data ?? {};
  const [txtSearch, setTxtSearch] = useState('');
  const txtSearchDebounced = useDebounced(txtSearch, 500);
  const [startDate, setStartDate] = useState<Date | null>(null);
  const [endDate, setEndDate] = useState<Date | null>(null);
  const endDatePicker = useRef<ReactDatePicker | null>(null);
  const [isOpenModal, setOpenModal] = useState(false);
  const [reportDetail, setReportDetail] = useState<IPostAndWFH>();

  const handleStartDateChange = (date: Date) => {
    setStartDate(date);
    if (!date) {
      setEndDate(null);
    } else {
      if (endDatePicker.current) {
        endDatePicker.current.setFocus();
      }
      if (endDate && date >= endDate) {
        setEndDate(null);
      }
    }
  };

  const handleEndDateChange = (date: Date) => {
    if (date && startDate && date >= startDate) {
      setEndDate(date);
    }
  };

  useEffect(() => {
    if (startDate && endDate && startDate !== endDate) {
      setFilter((filter) => ({
        ...filter,
        startDate: formatDate(startDate, 'MM/dd/yyyy'),
        endDate: formatDate(endDate, 'MM/dd/yyyy'),
      }));
    } else {
      setFilter((filter) => ({
        ...filter,
        startDate: null,
        endDate: null,
      }));
    }
  }, [startDate, endDate]);

  const onAction = (report: IPostAndWFH) => () => {
    setReportDetail(report);
    setOpenModal(true);
  };

  const onCloseModal = () => {
    setOpenModal(false);
  };

  const wfhColumns = useMemo(
    () =>
      [
        columnHelper.accessor('email', {
          id: 'email',
          enableSorting: true,
          sortDescFirst: true,
          header: () => <Box>Email address</Box>,
          cell: (info) => <Box>{info.getValue()}</Box>,
        }),
        columnHelper.accessor('totalDays', {
          id: 'totalDays',
          enableSorting: true,
          header: 'Number of requests for WFH',
          cell: (info) => info.getValue(),
        }),
        columnHelper.accessor('totalMissingPosts', {
          id: 'totalMissingPosts',
          header: 'Number of missing posts',
          cell: (info) => {
            const count = info.getValue();
            return (
              <Text
                color={count && count > 0 ? 'red' : ''}
                size="sm"
                fontWeight={count && count > 0 ? 'medium' : 'normal'}
              >
                {count}
              </Text>
            );
          },
        }),
        columnHelper.display({
          id: 'actions',
          enableSorting: false,
          header: () => <Center w="full">Actions</Center>,
          cell: (info) => (
            <Center onClick={(e) => e.stopPropagation()}>
              <RowAction onViewDetails={onAction(info.row.original)} />
            </Center>
          ),
        }),
      ] as ColumnDef<IPostAndWFH>[],
    [columnHelper]
  );

  const currentPage = useMemo(() => {
    const { skipCount, maxResultCount } = filter;
    return (maxResultCount + skipCount) / maxResultCount;
  }, [filter]);

  useEffect(() => {
    const { id, desc } = sorting?.[0] ?? {};
    const sort = `${id} ${desc ? SortDirection.desc : SortDirection.asc}`;

    setFilter((filter) => ({
      ...filter,
      sorting: sort,
      skipCount: 0,
    }));
  }, [sorting]);

  const onPageChange = (page: number) => {
    setFilter((filter) => ({
      ...filter,
      skipCount: filter.maxResultCount * (page - 1),
    }));
  };

  const onPageSizeChange = (pageSize: number) => {
    setFilter((filter) => ({
      ...filter,
      maxResultCount: pageSize,
      skipCount: 0,
    }));
  };

  useEffect(() => {
    setFilter((filter) => ({
      ...filter,
      keySearch: txtSearchDebounced,
    }));
  }, [txtSearchDebounced]);

  const exportData = wfhList
    .filter((item) => item.totalMissingPosts !== 0)
    .map((item) => ({
      email: item.email,
      totalDays: item.totalDays,
      totalMissingPosts: item.totalMissingPosts,
    }));

  return (
    <>
      <Box width={'100%'}>
        <HStack
          w="100%"
          display="flex"
          flexDirection={['column', 'column', 'column', 'row', 'row']}
          justifyContent={['flex-start', 'space-between']}
          alignItems={['flex-start']}
          pl={{ base: '24px' }}
        >
          <InputGroup
            alignItems={'center'}
            w={['90%', '80%', '70%', '40%', '30%', '30%']}
          >
            <Input
              type="text"
              placeholder="Enter email"
              fontSize="14px"
              onChange={(e) => setTxtSearch(e.target.value)}
            />
            <InputRightElement width="40px">
              <TbSearch />
            </InputRightElement>
          </InputGroup>
          <Box w={['80%', '80%', '70%', '50%', '40%', '30%']}>
            <DateRangePicker
              startDate={startDate}
              endDate={endDate}
              handleStartDateChange={handleStartDateChange}
              handleEndDateChange={handleEndDateChange}
              endDatePicker={endDatePicker}
            />
          </Box>
        </HStack>
        <Wrap pt={'12px'} pr={'16px'} justify="flex-end">
          <WrapItem>
            <div className={styles.btnExport}>
              <Button
                isDisabled={isLoading || isRefetching}
                onClick={() =>
                  handleExportExcelFile({ exportData: exportData, type: 'WFH' })
                }
              >
                <FaDownload />
              </Button>
            </div>
          </WrapItem>
          <WrapItem>
            <IconButton
              isDisabled={isLoading || isRefetching}
              isRound={true}
              variant="solid"
              aria-label="Done"
              fontSize="20px"
              icon={<AiOutlineReload />}
              onClick={() => refetch()}
            />
          </WrapItem>
        </Wrap>

        {isLoading || isRefetching ? (
          <Center h="200px">
            <Spinner mx="auto" speed="0.65s" thickness="3px" size="xl" />
          </Center>
        ) : (
          <EmptyWrapper
            isEmpty={!wfhList.length}
            h="200px"
            fontSize="xs"
            message={'No request found!'}
            display={'flex'}
            justifyContent={'center'}
            alignItems={'center'}
          >
            <Box
              p={{ base: '20px 24px' }}
              overflowX={'scroll'}
              w={{
                base: '100vw',
                lg: `calc(100vw - ${sideBarWidth}px)`,
              }}
            >
              <Table
                columns={wfhColumns}
                onRowClick={onAction}
                data={wfhList}
                onSortingChange={setSorting}
                sorting={sorting}
                onRowHover={true}
              />
            </Box>
          </EmptyWrapper>
        )}
        <HStack
          p="20px 30px 20px 30px"
          justifyContent="space-between"
          flexWrap="wrap"
        >
          <HStack alignItems="center" spacing="6px" flexWrap="wrap">
            <PageSize
              noOfRows={noOfRows}
              onChange={onPageSizeChange}
              defaultValue={+noOfRows[2].value}
            />
            <Spacer w="12px" />
            <ShowingItemText
              skipCount={filter.skipCount}
              maxResultCount={filter.maxResultCount}
              totalCount={totalCount}
            />
          </HStack>
          <Pagination
            total={totalCount}
            pageSize={filter.maxResultCount}
            current={currentPage}
            onChange={onPageChange}
            hideOnSinglePage
          />
        </HStack>

        {reportDetail && (
          <DetailModal
            isOpen={isOpenModal}
            onClose={onCloseModal}
            reportDetail={reportDetail}
            startDate={startDate}
            endDate={endDate}
          />
        )}
      </Box>
    </>
  );
};
