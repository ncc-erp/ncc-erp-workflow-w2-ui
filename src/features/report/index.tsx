import Page from 'common/components/Page';
import { TablePostAndWFH } from './components/PostAndWFHTable';
import { useUserPermissions } from 'hooks/useUserPermissions';
import { useMemo } from 'react';
import { Permissions } from 'common/constants';
import NotFound from 'common/components/NotFound';
import { useMediaQuery } from 'hooks/useMediaQuery';
import { useSetAppConfig } from 'stores/appConfig';
import { Box, Heading, Icon, IconButton, Image } from '@chakra-ui/react';
import { HiMenuAlt2 } from 'react-icons/hi';
import Logo from 'assets/images/ncc_logo.png';

const PostAndWFH = () => {
  const { hasPermission } = useUserPermissions();
  const isLargeScreen = useMediaQuery('(min-width: 1024px)');
  const { onOpenSideBar } = useSetAppConfig();
  const isHasPermission = useMemo(
    () => hasPermission(Permissions.VIEW_WFH_REPORTS),
    [hasPermission]
  );

  return isHasPermission ? (
    <Page>
      {!isLargeScreen && (
        <Box
          cursor="pointer"
          px="16px"
          pt={'20px'}
          display={'flex'}
          alignItems={'center'}
          gap={'16px'}
        >
          <IconButton
            size={{ base: 'md', md: 'sm' }}
            aria-label=""
            onClick={onOpenSideBar}
            rounded={{ base: 'md', md: 0 }}
            variant="ghost"
            bgColor="whiteAlpha.100"
            backdropFilter="auto"
            backdropBlur="4px"
            border={{ base: '1px', md: 0 }}
            borderColor="gray.100"
            aspectRatio="1/1"
          >
            <Icon fontSize="xl" as={HiMenuAlt2} />
          </IconButton>
          <Box display={'flex'} gap={'8px'}>
            <Image h="24px" src={Logo} />
            <Heading fontSize="18px">NCC Workflow</Heading>
          </Box>
        </Box>
      )}
      <Page.Header
        paddingBottom={isLargeScreen ? undefined : '0px'}
        height={isLargeScreen ? undefined : '60px'}
        margin={isLargeScreen ? undefined : '0px'}
      >
        <Page.HeaderLeft>
          <Page.Heading>Report WFH</Page.Heading>
        </Page.HeaderLeft>
        <Page.HeaderRight />
      </Page.Header>
      <Page.Body>
        <TablePostAndWFH />
      </Page.Body>
    </Page>
  ) : (
    <NotFound />
  );
};

export default PostAndWFH;
