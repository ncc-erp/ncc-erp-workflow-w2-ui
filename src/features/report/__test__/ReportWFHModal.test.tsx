import { render, screen } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { DetailModal } from '../components/DetailModal';
import { IPostAndWFH } from 'models/report';

describe('WFH Modal Detail', () => {
  const reportDetail: IPostAndWFH = {
    email: 'bob@example.com',
    totalDays: 7,
    totalMissingPosts: 3,
    posts: [
      {
        id: 1296,
        title: {
          rendered: 'Cùng khám phá Kỹ năng giao tiếp cơ bản',
        },
        status: 'publish',
        date: '2023-08-24T17:30:00',
        link: 'https://ant.ncc.asia/cung-kham-pha-ky-nang-giao-tiep-co-ban/',
      },
      {
        id: 1299,
        title: {
          rendered:
            '​Linkedin Tips &#8211; 5 cách tiếp cận để làm Linkedin nổi bật',
        },
        status: 'publish',
        date: '2023-08-14T17:30:00',
        link: 'https://ant.ncc.asia/5-cach-tiep-can-de-lam-linkedin-noi-bat/',
      },
      {
        id: 1301,
        title: {
          rendered: '10 lời khuyên cho hồ sơ LinkedIn chuyên nghiệp',
        },
        status: 'publish',
        date: '2023-08-24T17:30:00',
        link: 'https://ant.ncc.asia/10-loi-khuyen-cho-ho-so-linkedin-chuyen-nghiep/',
      },
      {
        id: 1303,
        title: {
          rendered: 'Cùng khám phá thế giới drama văn phòng',
        },
        status: 'publish',
        date: '2023-08-24T17:30:00',
        link: 'https://ant.ncc.asia/cung-kham-pha-the-gioi-drama-van-phong/',
      },
      {
        id: 1305,
        title: {
          rendered: 'CAREER MAP: TIẾP CẬN CHIẾN LƯỢC MỤC TIÊU NGHỀ NGHIỆP',
        },
        status: 'publish',
        date: '2023-08-24T17:30:00',
        link: 'https://ant.ncc.asia/career-map-tiep-can-chien-luoc-muc-tieu-nghe-nghiep/',
      },
    ],
    requests: [
      {
        workflowInstanceId: '3a0e2f0f-10ea-320d-f2da-7dc7be3e7688',
        input: {
          CurrentOffice: 'HN1',
          Project: 'hr',
          Reason: 'test',
          Dates:
            '11/10/2023,12/10/2023,14/08/2023,03/08/2023,21/09/2023,10/08/2023',
        },
        tenantId: '',
        creationTime: '2023-10-11T03:57:53.743222Z',
        creatorId: '3a059e91-b74f-f2a3-44eb-0a2863c0f138',
        id: '3a0e2f0f-20ce-b8ab-0db4-92149f136a67',
      },
      {
        workflowInstanceId: '3a0e2fdf-10ae-7350-0569-b37395c09005',
        input: {
          CurrentOffice: 'HN1',
          Project: 'REC',
          Reason: 'eqwwqe',
          Dates: '20/08/2023',
        },
        tenantId: '',
        creationTime: '2023-10-11T07:45:06.217695Z',
        creatorId: '3a059e91-b74f-f2a3-44eb-0a2863c0f138',
        id: '3a0e2fdf-24a8-88fb-2816-4881ce9a4dd2',
      },
    ],
    requestDates: [
      '03/08/2023',
      '10/08/2023',
      '14/08/2023',
      '20/08/2023',
      '21/09/2023',
      '11/10/2023',
      '12/10/2023',
    ],
  };

  const queryClient: QueryClient = new QueryClient();

  test('should match snapshot when rendering', () => {
    const { baseElement } = render(
      <QueryClientProvider client={queryClient}>
        <DetailModal
          isOpen={true}
          onClose={() => jest.fn()}
          reportDetail={reportDetail}
          startDate={null}
          endDate={null}
        />
      </QueryClientProvider>
    );
    expect(baseElement).toMatchSnapshot();
  })


  describe('WFH Modal Detail with the data above', () => {
    beforeEach(() => {
      render(
        <QueryClientProvider client={queryClient}>
        <DetailModal
          isOpen={true}
          onClose={() => jest.fn()}
          reportDetail={reportDetail}
          startDate={null}
          endDate={null}
        />
      </QueryClientProvider>
      );
    });

    it('Should have a Title when Report Details loaded.',() => {
      expect(screen.getAllByText(/Report Details/i)).toBeTruthy()
    })

    it('Should have a email when Report Details loaded.',() => {
      expect(screen.getAllByText(/bob@example.com/i)).toBeTruthy()
    })

    describe('Post Details attributes',() => {
      it('Should have a Post Details attributes when Report Detail loaded.', async() => {
        screen.findByText(/Post Details/i)
      })
    })  

    describe('Request Details attributes', () => {
      it('Should have a Request Details attributes when Report Detail loaded.', async() => {
        screen.findByText(/Request Details/i)
      })
    })
  })
});
