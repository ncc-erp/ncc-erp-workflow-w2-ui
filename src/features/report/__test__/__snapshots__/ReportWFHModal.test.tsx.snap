// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WFH Modal Detail 1`] = `
<body>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  />
  <div
    class="chakra-portal"
  >
    <div
      aria-hidden="true"
      class="chakra-modal__overlay css-13d1tkw"
      data-aria-hidden="true"
      style="opacity: 0;"
    />
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <div
        class="chakra-modal__content-container css-1oxhx2p"
        tabindex="-1"
      >
        <section
          aria-describedby="chakra-modal--body-:r0:"
          aria-labelledby="chakra-modal--header-:r0:"
          aria-modal="true"
          class="chakra-modal__content css-10m7e88"
          id="chakra-modal-:r0:"
          role="dialog"
          style="opacity: 0; transform: scale(0.95) translateZ(0);"
          tabindex="-1"
        >
          <header
            class="chakra-modal__header css-10d70kv"
            id="chakra-modal--header-:r0:"
          >
            <div
              class="chakra-stack css-1igwmid"
            >
              <img
                class="chakra-image css-kow2w3"
                src=""
              />
              <h2
                class="chakra-heading css-sigo6f"
              >
                <p
                  class="chakra-text css-1yfekn0"
                >
                  Report Details
                </p>
                <p
                  class="chakra-text css-issjhg"
                >
                  bob@example.com
                </p>
              </h2>
            </div>
          </header>
          <div
            class="chakra-modal__body css-0"
            id="chakra-modal--body-:r0:"
          >
            <hr
              aria-orientation="horizontal"
              class="chakra-divider css-vrx5g4"
            />
            <div>
              <div>
                <p
                  class="chakra-text css-yb4mb5"
                >
                  Post Details
                </p>
                <div
                  style="max-height: 450px; overflow-y: auto;"
                >
                  <div>
                    <div>
                      <div
                        class="css-1svpmll"
                      >
                        <p
                          class="chakra-text css-dynqen"
                        >
                          Title
                        </p>
                        <p
                          class="chakra-text css-dynqen"
                        >
                          Cùng khám phá Kỹ năng giao tiếp cơ bản
                        </p>
                        <p
                          class="chakra-text css-dynqen"
                        >
                          Date
                        </p>
                        <p
                          class="chakra-text css-dynqen"
                        >
                          24-08-2023 17:30
                        </p>
                        <div
                          class="undefined"
                        >
                          <div>
                            →
                          </div>
                        </div>
                      </div>
                    </div>
                    <div>
                      <div
                        class="css-1svpmll"
                      >
                        <p
                          class="chakra-text css-dynqen"
                        >
                          Title
                        </p>
                        <p
                          class="chakra-text css-dynqen"
                        >
                          10 lời khuyên cho hồ sơ LinkedIn chuyên nghiệp
                        </p>
                        <p
                          class="chakra-text css-dynqen"
                        >
                          Date
                        </p>
                        <p
                          class="chakra-text css-dynqen"
                        >
                          24-08-2023 17:30
                        </p>
                        <div
                          class="undefined"
                        >
                          <div>
                            →
                          </div>
                        </div>
                      </div>
                    </div>
                    <button>
                      Show more...
                    </button>
                  </div>
                </div>
              </div>
              <div>
                <p
                  class="chakra-text css-yb4mb5"
                >
                  Request Details
                </p>
                <div>
                  <p
                    class="chakra-text undefined css-71gjlu"
                  >
                    03/08/2023
                  </p>
                  <p
                    class="chakra-text undefined css-71gjlu"
                  >
                    10/08/2023
                  </p>
                  <p
                    class="chakra-text undefined css-71gjlu"
                  >
                    14/08/2023
                  </p>
                  <p
                    class="chakra-text undefined css-71gjlu"
                  >
                    20/08/2023
                  </p>
                  <p
                    class="chakra-text css-71gjlu"
                  >
                    21/09/2023
                  </p>
                  <p
                    class="chakra-text css-71gjlu"
                  >
                    11/10/2023
                  </p>
                  <p
                    class="chakra-text css-71gjlu"
                  >
                    12/10/2023
                  </p>
                </div>
              </div>
            </div>
          </div>
          <button
            aria-label="Close"
            class="chakra-modal__close-btn css-6sdaxy"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-onkibi"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                fill="currentColor"
              />
            </svg>
          </button>
        </section>
      </div>
    </div>
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</body>
`;
